charmdir.go

```diff
@@ -438,17 +444,11 @@ func (dir *CharmDir) MaybeGenerateVersionString() (string, string, error) {
 	// version string value is written to stdout if successful.
 	out, err := cmd.Output()
 	if err != nil {
+		logger.Warningf(
+			"%q version string generation failed : %v\nThis means that the charm version won't show in juju status.",
+			vcsType, err)
 		return "", vcsType, err
 	}
 	output := string(out)
-	if err == nil {
-		return output, vcsType, nil
-	}
-
-	// If there's an error, we may be able to get a relevant cause string.
-	exitErr, ok := err.(*exec.ExitError)
-	if !ok || len(exitErr.Stderr) == 0 {
-		return "", vcsType, err
-	}
-	return "", vcsType, fmt.Errorf("%s: %s", err, string(exitErr.Stderr))
```

Comment:
Good catch, this didn't do anything. I wonder why the deadcode didn't pick this up...?

Reply:
good questions... My ide did show that pretty hard :D